src/extension/importer/core/src/main/java/org/geoserver/importer/Importer.java

```diff
@@ -1001,8 +1002,9 @@ void doIndirectImport(ImportTask task) throws IOException {
                     // the result of a transform. there may be another way...
                     FeatureTypeInfo resource = getCatalog().getResourceByName(
                             featureType.getQualifiedName(), FeatureTypeInfo.class);
-                    if (resource.getNativeBoundingBox().isEmpty()
-                            || resource.getMetadata().get("recalculate-bounds").equals(Boolean.TRUE)) {
+                    if (resource.getNativeBoundingBox() == null || resource.getNativeBoundingBox().isEmpty()
+                            || Boolean.TRUE.equals(resource.getMetadata().get("recalculate-bounds"))
+                            || "true".equals(resource.getMetadata().get("recalculate-bounds"))) {
```

Comment:
Given that it is in two places, a method is probably a good idea. The functionality is already integration tested in ImportDataTest (see changes in #1247), but a more thorough test of just the method is probably a good idea.